### DISCUSSION TITLE
关于SMART AM40和友善RK3399 NanoPC-T4

### DISCUSSION THREAD

**Discussion by BackT0TheFuture** [ophub/amlogic-s9xxx-armbian#1465] [Rockchip] [answered]
首先感谢大佬做出的做贡献，

我尝试了目前repo中为am40编译的镜像，hdmi显示时文字总闪动且分辨率为1024x768，

显示器看长了眼确实不舒服，当然如果作为无界面服务用是没问题的，如何才能改善这种情况？

本人为码农，硬件菜鸟，现在想学习一下镜像制做，有以下疑问，还请有时间的各位前辈解答一下。

在AM40连上ttl后，可以看出以下如图输出

![ys](https://github.com/ophub/amlogic-s9xxx-armbian/assets/10088733/7b636c61-ab01-479f-a718-0eca208ea057)

通过Model里的名称我搜索到了友善一款板子叫 [NanoPC-T4](https://wiki.friendlyelec.com/wiki/index.php/NanoPC-T4/zh#.E5.BF.AB.E9.80.9F.E5.85.A5.E9.97.A8)，除了一些外设接口两款板子整体上很像，

am40目前能用的镜像很少，友善提供了较丰富的android和linux镜像（如下图），

其中有些带桌面且支持了硬件加速，并且提供相应的源码，

![dm](https://github.com/ophub/amlogic-s9xxx-armbian/assets/10088733/ceb12512-67f3-4767-bca6-f002c97bba3b)

所以本硬件小白想请教以下，怎么样才可以把友善提供的镜像或硬加速的之类的特性用于AM40这块板子上来，请大佬们说一下大致流程。

谢谢！！

**Comment by ophub** [comment, marked as answer]
这个u-boot是从用户分享的固件里复制的。从首页点AM40的名字可以链接到对应的测试帖子里。
你可以试试友善镜像能否正常使用。

**Comment by BackT0TheFuture** [comment]
@ophub 
感谢大佬百忙之路给予回复，咱repo下am40那个issue我关注过了的。

在上面提到的友善的镜像中下载后，都是分区的（如下图），不像repo中release中的是一个文件，

像这种情况我怎么样去刷它的镜像呢？

有需要我可以配合大佬做测试或提供板子的远程访问权限，

![pa](https://github.com/ophub/amlogic-s9xxx-armbian/assets/10088733/6179d415-bf7f-4f8d-8495-68b54d1e0c83)

谢谢！

**Comment by ophub** [comment]
https://wiki.friendlyelec.com/wiki/index.php/NanoPC-T4/zh

友善官网看说明

**Comment by BackT0TheFuture** [comment]
@ophub 
好的，大神！，我去仔细研究它的wiki。

当前刷am40时仅仅只有bootloader和system，

![am40](https://github.com/ophub/amlogic-s9xxx-armbian/assets/10088733/ccfa3ab4-422a-418e-a057-e3fb3b070b08)

system我是不是可以直接用友善system的img去刷呢

感谢解答！

**Reply by xiangtailiang** [reply]
不可以，你刷的omv是整合包，从0地址开始写没错。但友善的不是整合包，system是要写在对应的起始位置，你要按照友善的官方文档来刷，把所有分区都刷进去。
